src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,90 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private Node<K, V>[] table;
+    private int size = 0;
+
+    private class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getIndex(key);
+        Node<K, V> newNode = new Node<>(key, value, null);
+        Node<K, V> current = table[index];
+        if (current == null) {
+            table[index] = newNode;
+            size++;
+            return;
+        }
+        while (current != null) {
+            if (current.key == null && key == null
+                    || current.key != null && current.key.equals(key)) {
+                current.value = value;
+                return;
+            }
+            if (current.next == null) {
+                current.next = newNode;
+                size++;
+                break;
+            }
+            current = current.next;
+        }
+        if (size >= table.length * LOAD_FACTOR) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> current = table[index];
+        while (current != null) {
+            if (current.key == null && key == null
+                    || current.key != null && current.key.equals(key)) {
+                return current.value;
+            }
+            current = current.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode() % table.length);
+    }
+
+    @SuppressWarnings("unchecked")
+    private void resize() {
+        Node<K, V>[] oldTable = table;
+        table = new Node[oldTable.length * 2];
```

Comment:
Diamond operator should be used properly to avoid raw type usage. Use `new Node<K,V>[oldTable.length * 2]` instead of `new Node[oldTable.length * 2]` to ensure type safety.

Reply:
please follow AI's advice

Reply:
![2024-08-13 08_12_21-jv-hash-map – MyHashMap java](https://github.com/user-attachments/assets/f7e15efa-7eb1-421f-ac3d-18cfde8773fe)